hyperspy/signal.py

```diff
@@ -3284,6 +3300,7 @@ def get_dynamic_explorer_wrapper(*args, **kwargs):
                     " \"slider\", None, a Signal instance")
 
         self._plot.plot(**kwargs)
+        self.events.data_changed.connect(self.update_plot)
```

Comment:
This should be moved to `__init__()`, since it's never disconnected, and `update_plot()` checks for a valid plot.